src/java.base/share/classes/jdk/internal/misc/X-ScopedMemoryAccess.java.template

```diff
@@ -406,10 +406,19 @@ public class ScopedMemoryAccess {
                 scope.checkValidState();
             }
 
-            return VectorSupport.load(vmClass, e, length,
-                    BufferAccess.bufferBase(bb), BufferAccess.bufferAddress(bb, offset),
-                    bb, offset, s,
-                    defaultImpl);
+            byte[] base = (byte[]) BufferAccess.bufferBase(bb);
+
+            if (base == null) {
+              return VectorSupport.load(vmClass, e, length,
+                      null, BufferAccess.bufferAddress(bb, offset),
+                      bb, offset, s,
+                      defaultImpl);
+            } else {
+              return VectorSupport.load(vmClass, e, length,
+                      base, BufferAccess.bufferAddress(bb, offset),
+                      bb, offset, s,
+                      defaultImpl);
+            }
```

Comment:
Can we try collapsing that to:
```suggestion
              return VectorSupport.load(vmClass, e, length,
                      base, BufferAccess.bufferAddress(bb, offset),
                      bb, offset, s,
                      defaultImpl);
```
? That appeared to be ok when i ran the benchmark for the non-polluted cases.

Reply:
When I reverted changes I took base as `Object`, from `BufferAccess`, and did not work. Looks like cast to byte[] implies some kind of null checks, and we can use this.